src/cm/smcontroller/smcontroller.hpp

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright (C) 2025 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_CM_SMCONTROLLER_SMCONTROLLER_HPP_
+#define AOS_CM_SMCONTROLLER_SMCONTROLLER_HPP_
+
+#include <condition_variable>
+#include <memory>
+#include <mutex>
+
+#include <grpcpp/server.h>
+#include <servicemanager/v5/servicemanager.grpc.pb.h>
+
+#include <core/cm/imagemanager/itf/blobinfoprovider.hpp>
+#include <core/cm/launcher/itf/envvarhandler.hpp>
+#include <core/cm/launcher/itf/instancerunner.hpp>
+#include <core/cm/launcher/itf/monitoringprovider.hpp>
+#include <core/cm/networkmanager/itf/nodenetwork.hpp>
+#include <core/cm/nodeinfoprovider/itf/sminforeceiver.hpp>
+#include <core/cm/smcontroller/itf/logprovider.hpp>
+#include <core/cm/smcontroller/itf/smcontroller.hpp>
+#include <core/cm/unitconfig/itf/nodeconfighandler.hpp>
+#include <core/common/cloudconnection/itf/cloudconnection.hpp>
+#include <core/common/crypto/itf/certloader.hpp>
+#include <core/common/iamclient/itf/certprovider.hpp>
+
+#include "smhandler.hpp"
+
+namespace aos::cm::smcontroller {
+
+/**
+ * SM controller configuration.
+ */
+struct Config {
+    std::string mCMServerURL;
+    std::string mCertStorage;
+    std::string mCACert;
+};
+
+/**
+ * Service Manager Controller.
+ */
+class SMController : public SMControllerItf,
+                     public cloudconnection::ConnectionListenerItf,
```

Comment:
1. why some listeners are public other private?
2. aos core interface should be added to SMControllerItf 
3. update smcontroller design document 
4. group public and private interfaces.